supervisor/homeassistant/module.py

```diff
@@ -284,3 +297,57 @@ async def _hardware_events(self, device: Device) -> None:
             return
 
         self.sys_homeassistant.websocket.send_message({ATTR_TYPE: "usb/scan"})
+
+    async def backup(self, secure_tar_file: SecureTarFile) -> None:
+        """Backup Home Assistant Core config/ directory."""
+
+        # Let Home Assistant Core know we are about to backup
+        try:
+            await self.websocket.async_send_command({ATTR_TYPE: "backup/start"})
+
+        except HomeAssistantWSError:
+            _LOGGER.warning(
+                "Preparing backup of Home Assistant Core failed. Check HA Core logs."
+            )
+
+        def _write_tarfile():
+            with secure_tar_file as tar_file:
+                # Backup system
+                origin_dir = Path(self.sys_config.path_supervisor, FOLDER_HOMEASSISTANT)
+
+                # Backup data
+                atomic_contents_add(
+                    tar_file,
+                    origin_dir,
+                    excludes=HOMEASSISTANT_BACKUP_EXCLUDE,
+                    arcname=".",
+                )
+
+        try:
+            await self.sys_run_in_executor(_write_tarfile)
+        finally:
+            try:
+                await self.sys_homeassistant.websocket.async_send_command(
+                    {ATTR_TYPE: "backup/end"}
+                )
+            except HomeAssistantWSError:
+                _LOGGER.warning(
+                    "Error during Home Assistant Core backup. Check HA Core logs."
+                )
+
+    async def restore(self, secure_tar_file: SecureTarFile) -> None:
+        """Restore Home Assistant Core config/ directory."""
+
+        # Perform a restore
+        def _restore_tarfile():
+            origin_dir = Path(self.sys_config.path_supervisor, FOLDER_HOMEASSISTANT)
+
+            with secure_tar_file as tar_file:
+                tar_file.extractall(path=origin_dir, members=tar_file)
+
+        try:
+            _LOGGER.info("Restore Home Assistant Core config folder")
+            await self.sys_run_in_executor(_restore_tarfile)
+            _LOGGER.info("Restore Home Assistant Core config folder done")
+        except (tarfile.TarError, OSError) as err:
+            _LOGGER.warning("Can't restore Home Assistant Core config folder: %s", err)
```

Comment:
Shouldn't this raise? if it fails we don't want to continue with the restore?

Reply:
This replaces the behavior so far: `_folder_restore()` in `restore_folders()` uses the same exception handler. That said, I think backup as well as restore needs some love wrt error handing.